app.js

```diff
@@ -66,10 +66,10 @@ app.use(function(req, res, next) {
 });
 
 app.get('/validate', function (req, res, next) {
-  if (isValidToken(req.query.token)) {
-    sessionExists(req.query.token).then(function() {
+  var apiToken = req.query.token;
+  if (isValidToken(apiToken)) {
```

Comment:
my bad, i was looking at a single commit and not the whole set of changes...as you were.